drivers/gpu/drm/lima/lima_device.c

```diff
@@ -78,6 +79,31 @@ static void lima_clk_fini(struct lima_device *dev)
 	clk_disable_unprepare(dev->clk_bus);
 }
 
+static int lima_regulator_init(struct lima_device *dev)
+{
+	int ret;
+	dev->reg = devm_regulator_get_optional(dev->dev, "mali");
```

Comment:
I can see this regulator in RK DTS but not Exynos4 DTS, so you added it by yourself in your kernel tree, right?

Reply:
Yes. It's board-specific, but my board isn't in mainline yet.